types/bool.go

```diff
@@ -5,13 +5,21 @@ import (
 	"fmt"
 
 	"github.com/hashicorp/terraform-plugin-framework/attr"
+	"github.com/hashicorp/terraform-plugin-framework/diag"
+
 	"github.com/hashicorp/terraform-plugin-go/tftypes"
 )
 
 var (
-	_ attr.Value = Bool{}
+	_ BoolValuable = Bool{}
 )
 
+type BoolValuable interface {
```

Comment:
Let's ensure all these new `types` interfaces have Go documentation. 👍 

```suggestion
// BoolValuable extends attr.Value for boolean value types.
// Implement this interface to create a custom Bool value type.
type BoolValuable interface {
```